drivers/zfs/zfs.go

```diff
@@ -406,10 +406,20 @@ func (d *Driver) Get(id string, options graphdriver.MountOpts) (_ string, retErr
 		return "", errors.Wrap(err, "error creating zfs mount")
 	}
 
-	// this could be our first mount after creation of the filesystem, and the root dir may still have root
-	// permissions instead of the remapped root uid:gid (if user namespaces are enabled):
-	if err := os.Chown(mountpoint, rootUID, rootGID); err != nil {
-		return "", fmt.Errorf("error modifying zfs mountpoint (%s) directory ownership: %v", mountpoint, err)
+	readWrite := true
+	for _, option := range options.Options {
+		if option == "ro" {
+			readWrite = false
+			break
+		}
+	}
+
+	if readWrite {
+		// this could be our first mount after creation of the filesystem, and the root dir may still have root
+		// permissions instead of the remapped root uid:gid (if user namespaces are enabled):
+		if err := os.Chown(mountpoint, rootUID, rootGID); err != nil {
```

Comment:
I'd love to see a test for that if possible.

Reply:
I can try. Can you give me a hint how to do that?

Reply:
Hm, I cant find zfs integration tests in the current test-runs, but usually I would expect that we can create a new test case in one of (or a new) bats test files there: https://github.com/containers/storage/tree/master/tests

But if ZFS is not supported by the CI right now (which seems the case to me), then we should not block that PR here.